packages/cashier/src/Components/CashierDefault/cashier-default.jsx

```diff
@@ -15,76 +15,97 @@ const CashierDefault = ({
+        openRealAccountSignup('choose');
+        shouldNavigateAfterChooseCrypto(target);
+    };
+
+    const openTarget = target => {
+        setDepositTarget(target);
+        if (is_crypto) {
+            openRealAccount(target);
+        } else if (has_crypto_account) {
+            openRealAccount(target);
+        } else {
+            openRealAccountSignup('add');
         }
+    };
+
+    const fiatAccountConditions = (next_location, current_location) => {
         if (has_fiat_account) {
-            toggleAccountsDialog();
-            return;
+            shouldNavigateAfterPrompt(next_location, current_location);
+        } else {
+            openRealAccountSignup('deposit_cash');
         }
-        setManageRealAccountActiveTabIndex(1);
-        openRealAccountSignup('manage');
     };
 
-    const onClickDepositCrypto = () => {
+    const onClickDepositCash = () => {
+        setDepositTarget(routes.cashier_deposit);
+
         if (is_crypto) {
+            fiatAccountConditions(routes.cashier_deposit, 'deposit');
+        } else {
             setIsDeposit(true);
-            return;
-        }
-        if (has_crypto_account) {
-            toggleAccountsDialog();
-            return;
         }
-        setManageRealAccountActiveTabIndex(0);
-        openRealAccountSignup('manage');
+    };
+
+    const onClickDepositCrypto = () => {
+        openTarget(routes.cashier_deposit);
+    };
+
+    const onClickOnramp = () => {
+        openTarget(routes.cashier_onramp);
     };
 
     const onClickPaymentAgent = () => {
-        history.push(routes.cashier_pa);
+        setShouldShowAllAvailableCurrencies(true);
+        setDepositTarget(routes.cashier_pa);
+        openRealAccount(routes.cashier_pa);
     };
 
     const onClickDp2p = () => {
-        history.push(routes.cashier_p2p);
+        setDepositTarget(routes.cashier_p2p);
+
+        if (is_crypto) {
+            fiatAccountConditions(routes.cashier_p2p, 'DP2P');
+        } else {
+            history.push(routes.cashier_p2p);
+        }
     };
 
     const getDepositOptions = () => {
         const options = [];
         options.push(Providers.createCashProvider(onClickDepositCash));
-        if (!is_eu) {
-            options.push(Providers.createCryptoProvider(onClickDepositCrypto));
-
-            // Put the crypto option first in case the account is crypto.
-            if (is_crypto)
-                options.sort(
-                    (first_option, second_option) => options.indexOf(second_option) - options.indexOf(first_option)
-                );
-        }
+        options.push(Providers.createCryptoProvider(onClickDepositCrypto));
+        options.push(Providers.createOnrampProvider(onClickOnramp, is_crypto));
         if (is_payment_agent_visible) {
             options.push(Providers.createPaymentAgentProvider(onClickPaymentAgent));
         }
-        if (is_p2p_enabled) {
+        if (!is_eu && user_fiat_currency[0]?.currency === 'USD') {
```

Comment:
Why not just use the`is_p2p_enabled` prop here?

Reply:
@carol-binary Because `is_p2p_enabled` returns false in case the account is crypto, and here we only want to hide the option for the users who are in the forbidden countries.

Reply:
So we want to allow it for crypto accounts?

Reply:
If the user with crypto account click on the option, we will ask to switch the account to the fiat one or create a fiat account.

Reply:
I don't think this is the right approach though @Bahar-FS, we don't show DP2P for crypto accounts so it doesn't make sense to ask them to switch anyway, why not just show it for USD accounts?

Reply:
@carol-binary `Deposit with DP2P` section is available in the onboarding page irrespective of the currency (except  for restricted countries). Clicking on it will ask user to switch or create a fiat account to use DP2P if the active account is crypto. It is part of the onboarding flow.